### PR TITLE
refactor: move 2FA disable logic to Authorization service

### PR DESCRIPTION
**Pull request by Saraabodeeb**
Summary
Following the feedback to make PRs more atomic and focused, this PR refactors the 2fa:disable logic. It moves the core functionality from the Meteor methods to the Authorization service within core-services.

Changes
Interface Update: Added disable2FA(uid: string, code: string) to the IAuthorization interface in packages/core-services.

Service Implementation: Implemented the disable2FA logic in Authorization service.

Added security verification using TOTP.verify to ensure the 2FA code is valid before disabling.

Used Users.disable2FAByUserId(uid) for database updates.

Method Refactoring: Updated the 2fa:disable Meteor method in apps/meteor to act as a proxy, delegating the core logic to the Authorization service.

Testing: Added a basic unit test in packages/core-services/tests/Authorization.spec.ts to verify the method definition.

Security Note
[!IMPORTANT]
As discussed in previous reviews, security is a priority. The service now handles the verification step internally before performing any database operations to prevent unauthorized 2FA deactivation.

## Summary by CodeRabbit

* **Refactor**
  * Centralized the two-factor authentication (2FA) disable flow into the authorization layer for consistent behavior.
  * Disabling 2FA now reliably clears TOTP-related fields and emits user-change notifications when successful.
* **Tests**
  * Added comprehensive tests covering invalid users, already-disabled 2FA, invalid codes, and successful 2FA disable scenarios.

### CODE REVIEW

File: apps/meteor/server/services/authorization/service.ts [hunk 1]
```diff
@@ -202,4 +203,20 @@ export class Authorization extends ServiceClass implements IAuthorization {
 
 		return Roles.isUserInRoles(userId, roleIds, scope);
 	}
+	async disable2FA(uid: string, code: string): Promise<boolean> {
+		const user = await Users.findOneById(uid);
+		if (!user?.services?.totp?.enabled) return false;
```
Comment (KevLehman) [inline apps/meteor/server/services/authorization/service.ts]: 2 things:
- We prefer multiline ifs (`if () { ... }`), pls fix in other as well
- We used to have a `!user` check that throwed an error, that should be kept for compatibility

File: apps/meteor/app/2fa/server/methods/disable.ts [hunk 2]
```diff
@@ -1,55 +1,21 @@
...
+
+        const result = await Authorization.disable2FA(userId, code);
+
+        if (result) {
+            void notifyOnUserChange({ 
+                clientAction: 'updated', 
+                id: userId, 
+                diff: { 'services.totp.enabled': false } 
+            });
+        }
```
Comment (KevLehman) [inline apps/meteor/app/2fa/server/methods/disable.ts:33]: Check if this can be moved to the service without issues.

File: packages/core-services/tests/Authorization.spec.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { expect } from 'chai';
+import type { IAuthorization } from '../src/types/IAuthorization';
+
+describe('Authorization Service - 2FA', () => {
```
Comment (KevLehman) [inline packages/core-services/tests/Authorization.spec.ts]: The tests should test the actual behavior.

**Comment by Saraabodeeb** [comment]
Hi @KevLehman, I completely agree with testing the actual behavior! Since the 2fa:disable logic didn't have tests prior to this refactor, I want to make sure we set up the right testing foundation here.

To start, I've added a strict Behavioral Unit Test (apps/meteor/server/services/authorization/service.spec.ts) that uses calledOnceWithExactly to verify the exact parameters passed to DB operations and notifications, rather than just mocking return values.

I also want to add a proper Integration Test (inserting a real user into a Test DB and asserting the state change). However, to avoid MongoNotConnectedError in the CI, could you point me to the standard directory for DB-connected core service tests? (e.g., apps/meteor/tests/integration/...?).

Once you point me to the right suite, I’ll happily add the full integration test! Let me know what you think.

File: apps/meteor/server/services/authorization/service.ts [hunk 4]
```diff
@@ -202,4 +206,44 @@ export class Authorization extends ServiceClass implements IAuthorization {
...
+		});
+
+		if (!verified) {
+			throw new Meteor.Error('invalid-totp');
+		}
+
+		const { modifiedCount } = await Users.disable2FAByUserId(uid);
+
+		if (modifiedCount > 0) {
+			void notifyOnUserChange({
```
Comment (KevLehman) [inline apps/meteor/server/services/authorization/service.ts]: curious: can u confirm this works when running microservices?

You can run them locally with `yarn ms` and then going to `localhost:4000`. if it doesn't work, this would need to be moved back to the meteor method.

**Comment by Saraabodeeb** [comment]
Hi @KevLehman, that was a brilliant catch regarding the microservices architecture!

Rather than risking a silent failure in isolated environments, I took the safer architectural route you suggested. I've moved the notifyOnUserChange call back to the Meteor Method (disable.ts), so the Authorization service remains pure and handles only the DB logic.

I also went ahead and addressed the automated review feedback:

Swapped Meteor.Error for MeteorError to ensure full microservices compatibility.

Used unset in the notification payload to correctly strip the cached totp fields on the client.

Added edge-case test coverage for these changes.

Note on CI: It looks like the Dionisio QA check is currently failing because the PR is missing the stat: QA assured label and a milestone. Could you or one of the maintainers kindly apply them so the workflows can proceed?

**Comment by Saraabodeeb** [comment]
Hi @KevLehman, hope you're having a good week! Just gently bumping this PR. I addressed the microservices feedback a while ago, and I've just updated the branch to keep it sync with develop. Could you kindly take a quick look when you have a moment? Thank you!
